docs/sync.js

```diff
@@ -242,6 +242,59 @@ class User {
      * }
      */
     retrieveAccount(provider, username) {}
+
+    /**
+     * Asynchronously retrieve all permissions associated with the user calling this method.
+     * @param {string} recipient the optional recipient of the permission. Can be either
+     * 'any' which is the default, or 'currentUser' or 'otherUser' if you want only permissions
+     * belonging to the user or *not* belonging to the user.
+     * @returns {Results} a queryable collection of permission objects that provide detailed 
+     * information regarding the granted access.
+     * The collection is a live query similar to what you would get by callig Realm.objects,
+     * so the same features apply - you can listen for notifications or filter it.
+     */
+    getGrantedPermissions(recipient) { }
+
+    /**
+     * Changes the permissions of a Realm.
+     * @param {string} condition - A condition that will be used to match existing users against.
+     * This should be an object, containing either the key 'userId', or 'metadataKey' and 'metadataValue'.
+     * @param {string} realmUrl - The path to the realm that you want to apply permissions to.
+     * @param {string} accessLevel - The access level you want to set: 'none', 'read', 'write' or 'admin'.
+     * @returns {Promise} a Promise that, upon completion, indicates that the permissions have been 
+     * successfully applied by the server. It will be resolved with the PermissionChange object that refers
+     * to the applied permission.
+     */
+    applyPermissions(condition, realmUrl, accessLevel) { }
+
+    /**
+     * Generates a token that can be used for sharing a Realm.
+     * @param {string} realmUrl - The Realm URL whose permissions settings should be changed. Use * to change 
+     * the permissions of all Realms managed by this user.
+     * @param {string} accessLevel - The access level to grant matching users. Note that the access level 
+     * setting is absolute, i.e. it may revoke permissions for users that previously had a higher access level. 
```

Comment:
My understanding is that the `offerPermissions` access level setting is complimentary rather than absolute.

Reply:
This is copied from your documentation of the .NET equivalent: https://github.com/realm/realm-dotnet/blob/master/Realm/Realm.Sync/User.cs#L405

Reply:
Oh, shoot. Filed an issue to fix it: https://github.com/realm/realm-dotnet/issues/1540